pkg/sql/conn_executor.go

```diff
@@ -3687,3 +3759,56 @@ func init() {
 		return planGistFromCtx(ctx)
 	})
 }
+
+type jobsInfo struct {
+	jobsCreated           jobsCollection
+	uniqueJobsToCreate    map[descpb.ID]*jobs.Record
+	nonUniqueJobsToCreate []*jobs.Record
+}
+
+func newJobsInfo() *jobsInfo {
+	ret := &jobsInfo{}
+	ret.reset()
+	return ret
+}
+
+func (j *jobsInfo) addCreatedJobID(jobID ...jobspb.JobID) {
+	j.jobsCreated.add(jobID...)
+}
+
+func (j *jobsInfo) addNonUniqueJobToCreate(jobRecord *jobs.Record) {
+	j.nonUniqueJobsToCreate = append(j.nonUniqueJobsToCreate, jobRecord)
+}
+
+func (j *jobsInfo) reset() {
+	j.jobsCreated.reset()
+	j.uniqueJobsToCreate = make(map[descpb.ID]*jobs.Record)
+	j.nonUniqueJobsToCreate = nil
+}
+
+func (j *jobsInfo) numJobsToCreate() int {
+	return len(j.uniqueJobsToCreate) + len(j.nonUniqueJobsToCreate)
+}
+
+func (j *jobsInfo) hasJobsToCreate() bool {
+	return j.numJobsToCreate() > 0
+}
+
+func (j *jobsInfo) forEachJobToCreate(fn func(jobRecord *jobs.Record) error) error {
+	for _, r := range j.uniqueJobsToCreate {
+		if err := fn(r); err != nil {
+			return err
+		}
+	}
+	for _, r := range j.nonUniqueJobsToCreate {
+		if err := fn(r); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (j *jobsInfo) releaseJobsToRecreate() {
+	j.uniqueJobsToCreate = make(map[descpb.ID]*jobs.Record)
```

Comment:
same here about clearing the map as opposed to allocating a new one

Reply:
done.